src/ip/QosConfig.cc

```diff
@@ -465,64 +467,61 @@ Ip::Qos::Config::parseConfigLine()
  * which means no StoreEntry references. Just a basic char* buffer.
 */
 void
-Ip::Qos::Config::dumpConfigLine(char *entry, const char *name) const
+Ip::Qos::Config::dumpConfigLine(std::ostream &os, const char *name) const
 {
-    char *p = entry;
     if (isHitTosActive()) {
-
-        p += snprintf(p, 11, "%s", name); // strlen("qos_flows ");
-        p += snprintf(p, 4, "%s", "tos");
+        os << name << " tos";
 
         if (tosLocalHit > 0) {
-            p += snprintf(p, 16, " local-hit=0x%02X", tosLocalHit);
+            os << " local-hit=" << AsHex(tosLocalHit);
         }
         if (tosSiblingHit > 0) {
-            p += snprintf(p, 18, " sibling-hit=0x%02X", tosSiblingHit);
+            os << " sibling-hit=" << AsHex(tosSiblingHit);
```

Comment:
The old code specified the width and, hence, printed leading zeros for single-digit hex values. If this PR should preserve that functionality, let's preserve it (e.g., by adding/using the corresponding AsHex method(s)). Otherwise, please note this output format change in the PR description.

Reply:
In practice it seems to be working, as shown in the test output.
But I agree it's a good idea to make sure it's not by accident, so I'll add that as well

Reply:
> In practice it seems to be working, as shown in the test output.

The test [output](https://github.com/squid-cache/squid/pull/1577#issuecomment-1802098544) shows that the current PR code is _not_ "working" (as far as this change request is concerned):

```
qos_flows mark local-hit=0x1 sibling-hit=0x2 parent-hit=0x3 ...
```